src/Security/Permission.php

```diff
@@ -392,10 +392,16 @@ public static function groupList($memberID = null)
      */
     public static function grant($groupID, $code, $arg = "any")
     {
-        $perm = new Permission();
-        $perm->GroupID = $groupID;
-        $perm->Code = $code;
-        $perm->Type = self::GRANT_PERMISSION;
+        $permissions = Permission::get()->filter(['GroupID' => $groupID, 'Code' => $code]);
+        
+        if ($permissions && $permissions->count() > 0) {
+            $perm = $permissions->first();
+        } else {
+            $perm = new Permission();
+            $perm->GroupID = $groupID;
+            $perm->Code = $code;
+            $perm->Type = self::GRANT_PERMISSION;
```

Comment:
Move this after the condition - we need to make sure the permission is of type `self::GRANT_PERMISSION` in case it was previously denied.

Also make sure there is a test for this situation (and verify what happens currently without this PR if a permission is denied and then granted)